common/setups/hybrid/gmm_system.py

```diff
@@ -595,36 +595,44 @@ def recognition(
         self,
         name: str,
         iters: List[int],
+        lm_scales: List[int],
+        feature_scorer: Tuple[str, str],
+        optimize_am_lm_scale: bool,
+        # parameters just for passing through
         corpus: str,
         feature_flow: str,
-        feature_scorer,
-        pronunciation_scale: List[int],
-        lm_scale: List[int],
-        search_params: dict,
+        pronunciation_scale: int,
+        search_parameters: dict,
         rtf: int,
         mem: int,
         parallelize_conversion: bool,
         lattice_to_ctm_kwargs: dict,
         **kwargs,
     ):
+        """
+
+        :param name:
+        :param iters:
+        :param lm_scales:
+        :param feature_scorer: (training_corpus_name, training_name)
+        :param optimize_am_lm_scale:
+        :param kwargs: see meta.System.recog and meta.System.recog_and_optimize
+        :return:
+        """
+        assert "lm_scale" not in kwargs, "please use lm_scales for GmmSystem.recognition()"
         with tk.block(f"{name}_recognition"):
-            optimize_am_lm_scale = (
-                self.gmm_monophone_args.monophone_recognition_args.pop(
-                    "optimize_am_lm_scale", False
-                )
-            )
             recog_func = self.recog_and_optimize if optimize_am_lm_scale else self.recog
 
-            for it in iters:
+            for it, l in itertools.product(iters, lm_scales):
                 recog_func(
-                    name=f"{name}-{it:02d}",
-                    prefix=f"recog_{corpus}_{name}/",
+                    name=f"iter{it:02d}-lm{l}",
```

Comment:
`pronunciation_scale` if set, does not go into the alias/output name.

Reply:
I did this on purpose, because if we add pronunciation_scale, we could discuss other stuff as well. I only added iteration and lm here, because this is stuff that is handled separately here. Also at the top I added this "only for passing through" comment. So everything behind there should be named specifically by the user.